src/object/cli_obj.c

```diff
@@ -5598,8 +5601,11 @@ shard_anchors_check_alloc_bufs(struct obj_auxi_args *obj_auxi,
 			req_tgts->ort_shard_tgts[i].st_rank = DAOS_TGT_IGNORE;
 		}
 
-		if (sub_anchor->ssa_recxs != NULL)
-			continue;
+		if (sub_anchor->ssa_recxs != NULL) {
+			if (sub_anchors->sa_nr == nr)
+				continue;
+			D_FREE(sub_anchor->ssa_recxs);
```

Comment:
Do you still need to address the issue of init/fini to deallocate the sub_anchors when the iteration terminates early?

Reply:
oh, I will resolve that in a different patch.